### PR TITLE
Maintain Account Number of Original Queue

### PR DESCRIPTION
**Pull request by dpbackes**
- The previous implementation assumed that the DLQ was on the same
  account as `overdrive` was running under but that might not always be
  the case. This makes it so that we preserve the original account number
  when we look up the DLQ

See https://github.com/articulate/team-slides/issues/262

### CODE REVIEW

**Review by SarahRiley — APPROVED** [review]
Looks good from the docker side!

File: src/overdrive.js [hunk 1]
```diff
@@ -1,19 +1,31 @@
...
+const getQAttributes = (opts) =>
   pipe(
     pick(['QueueUrl']),
     assoc('AttributeNames', ['All']),
     pipeP(
       promisify(sqs.getQueueAttributes, sqs),
       prop('Attributes')
+    ),
+    when(
+      () => opts.verbose,
```
Comment (mwinckler) [inline src/overdrive.js:21]:
```suggestion
      () => opts.verbose,
```

File: src/overdrive.js [hunk 2]
```diff
@@ -1,19 +1,31 @@
 const { combineP, promisify } = require('@articulate/funky')
 
 const {
-  assoc, last, objOf, pick, pipe, pipeP, prop, split
+  assoc, concat, objOf, pick, pipe, pipeP, prop, split, takeLast, tap, when, zipObj
```
Comment (mwinckler) [inline src/overdrive.js:4]:
```suggestion
  assoc, concat, pick, pipe, pipeP, prop, split, takeLast, when, zipObj
```

File: src/overdrive.js [hunk 3]
```diff
@@ -1,19 +1,31 @@
...
 } = require('ramda')
 
+const { tapP } = require('@articulate/funky')
+
 const redriveDLQ = require('./redriveDLQ')
 const sqs = require('./sqs')
 
-const getQAttributes =
+const getQAttributes = (opts) =>
```
Comment (mwinckler) [inline src/overdrive.js:12]:
```suggestion
const getQAttributes = opts =>
```

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+ARG NPM_TOKEN=secret
+
+RUN apt-get update -qq && apt-get install -y p7zip-full
+
+ENV npm_config_loglevel=warn
+
+COPY --chown=service:service package.json yarn.lock $SERVICE_ROOT/
+COPY --chown=service:service . $SERVICE_ROOT/
+
+RUN yarn install --pure-lockfile && yarn cache clean && npm link
```
Comment (mwinckler) [inline Dockerfile:11]: This doesn't work for me - `yarn` seems to have removed the `--pure-lockfile` arg in version 2. I believe it's now the default not to update the lockfile on `yarn install`.

```
...
Step 7/7 : RUN yarn install --pure-lockfile && yarn cache clean && npm link
 ---> Running in 18baaa390484
Unknown Syntax Error: Unsupported option name ("--pure-lockfile").
```

```suggestion
RUN yarn install && yarn cache clean && npm link
```

**Review by mwinckler — CHANGES_REQUESTED** [review]
The docker container doesn't seem to work for me; complains it can't find module `commander`:

```
mattw@mantis:~/source/articulate/overdrive$ art -on 360-stage docker-compose run --rm app bash
Building consul                               
Sending build context to Docker daemon  18.43kB
Step 1/5 : FROM consul:1.9.6
 ---> a567dca67fd8
Step 2/5 : COPY files/route_config /route_config
 ---> Using cache
 ---> db98579b6801
Step 3/5 : COPY files/entrypoint.sh /entrypoint.sh
 ---> Using cache
 ---> e6a4a6678a6e
Step 4/5 : RUN chmod +x /entrypoint.sh
 ---> Using cache
 ---> effe5dcf6718
Step 5/5 : ENTRYPOINT ["/entrypoint.sh"]
 ---> Using cache
 ---> 658cb535e5ad
Successfully built 658cb535e5ad
Successfully tagged skipper_consul:latest
skipper_consul_1 is up-to-date
skipper_registrator_1 is up-to-date
skipper_fabio_1 is up-to-date
Creating overdrive_app_run ... done
CONSUL_ADDR are not set skipping Consul exports
VAULT_ADDR is not set, skipping Vault exports
root@1cb5bd730d12:/service# overdrive -V 
internal/modules/cjs/loader.js:818
  throw err;
  ^

Error: Cannot find module 'commander'
Require stack:
- /service/src/cli.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:815:15)
    at Function.Module._load (internal/modules/cjs/loader.js:667:27)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/service/src/cli.js:3:17)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:60:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '/service/src/cli.js' ]
}
root@1cb5bd730d12:/service# 
```

File: Dockerfile (hunk 4 above)
Comment (mwinckler) [reply Dockerfile:11]: Disregard - this was due to a local `.yarn` on my machine from running yarn outside the container. Deleting that made this work fine.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,11 @@
+FROM articulate/articulate-node:12-stretch-slim
+ARG NPM_TOKEN=secret
+
+RUN apt-get update -qq && apt-get install -y p7zip-full
```
Comment (mwinckler) [inline Dockerfile:4]: Missed this earlier - is there a reason we need to install `p7zip-full`?

**Review by mwinckler — APPROVED** [review]
All those errors were due to a newer version of yarn on my machine being volumed in - this'll be good to go after addressing the linting. Output looks good!

File: Dockerfile (hunk 5 above)
Comment (dpbackes) [reply Dockerfile:4]: No, I copy/pasted this from `team-slides` and missed this in the cleanup

**Comment by dpbackes** [comment]
@mwinckler I also updated the package version

**Pull request merged**